keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/multiple/LambdaModelVertexTest.java

```diff
@@ -0,0 +1,231 @@
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.VertexLabel;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import io.improbable.keanu.vertices.intgr.IntegerVertex;
+import io.improbable.keanu.vertices.model.LambdaModelVertex;
+import io.improbable.keanu.vertices.model.ModelVertex;
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LambdaModelVertexTest {
+
+    /*
+    The model we are mimicking here lives inside a Java function, modelExecution.
+
+    It takes one input, Temperature, and produces two outputs of type double, Chance of Rain & Humidity. These outputs
+    are written to file.
+
+    It also produces one integer output and one boolean output. Suggested Factor of Suncream and 'is it sunny'. These
+    are also written to file.
+     */
+
+    private KeanuRandom random;
+    private DoubleVertex inputToModel = new ConstantDoubleVertex(50.);
+    private SimpleWeatherModel weatherModel;
+
+    @Before
+    public void mockFilesToReadModelOutput() throws IOException {
+        random = new KeanuRandom(1);
+        weatherModel = new SimpleWeatherModel(inputToModel);
+    }
+
+    @Test
+    public void canRunAModelInAModel() {
+        inputToModel = new ConstantDoubleVertex(25);
+        weatherModel.setInputToModel(inputToModel);
+        Map<VertexLabel, Vertex<? extends Tensor>> inputs = new HashMap<>();
+        inputs.put(new VertexLabel("Temperature"), inputToModel);
+
+        ModelVertex model = new LambdaModelVertex(inputs, this::modelExecution, this::updateValues);
+        DoubleVertex chanceOfRain = model.getDoubleModelOutputVertex(new VertexLabel("ChanceOfRain"));
+        DoubleVertex humidity = model.getDoubleModelOutputVertex(new VertexLabel("Humidity"));
+
+        DoubleVertex shouldIBringUmbrella = chanceOfRain.times(humidity);
+
+        double inputValue = 10.0;
+
+        inputToModel.setAndCascade(inputValue);
+        Assert.assertEquals(shouldIBringUmbrella.getValue().scalar(), 20.0, 1e-6);
+    }
+
+    @Test
+    public void canRunEvalOnTheOutputsToRecalculateTheModel() {
+        inputToModel = new ConstantDoubleVertex(25);
+        weatherModel.setInputToModel(inputToModel);
+        Map<VertexLabel, Vertex<? extends Tensor>> inputs = new HashMap<>();
+        inputs.put(new VertexLabel("Temperature"), inputToModel);
```

Comment:
one-liner:

        Map<VertexLabel, Vertex<? extends Tensor>> inputs = ImmutableMap.of(new VertexLabel("Temperature"), inputToModel);

Reply:
nice - updated